src/mlpack/methods/reinforcement_learning/replay/sumtree.hpp

```diff
@@ -0,0 +1,194 @@
+/**
+ * @file sumtree.hpp
+ * @author Xiaohong
+ *
+ * This file is an implementation of sumtree.
+ *
+ * reference:
+ * [1] https://github.com/openai/baselines/blob/master/baselines/common/segment_tree.py
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_SUMTREE_HPP
+#define MLPACK_METHODS_RL_SUMTREE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace rl {
+
+/**
+ * Implementation of SumTree.
+ *
+ * Build a Segment Tree like data structure.
+ * https://en.wikipedia.org/wiki/Segment_tree
+ *
+ * Used to maintain prefix-sum of an array.
+ *
+ */
+
```

Comment:
We can remove the empty line here. Also can you add a comment for the template parameter?